builder/vsphere/common/step_reattach_cdrom.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+//go:generate packer-sdc struct-markdown
+//go:generate packer-sdc mapstructure-to-hcl2 -type ReattachCDRomConfig
+
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+	"github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/driver"
+)
+
+type ReattachCDRomConfig struct {
+	// Reattach one or more configured CD-ROM devices. Range: 1-4.
+	// You can reattach up to 4 CD-ROM devices to the final build artifact.
+	// If set to 0, `reattach_cdroms` is ignored and the step is skipped.
+	// When set to a value in the range, `remove_cdrom` is ignored and
+	// the CD-ROM devices are kept without any attached media.
+	ReattachCDRom int `mapstructure:"reattach_cdroms"`
+}
+
+type StepReattachCDRom struct {
+	Config      *ReattachCDRomConfig
+	CDRomConfig *CDRomConfig
+}
+
+func (s *StepReattachCDRom) Run(_ context.Context, state multistep.StateBag) multistep.StepAction {
+	ui := state.Get("ui").(packersdk.Ui)
+	vm := state.Get("vm").(driver.VirtualMachine)
+
+	var err error
+
+	// Check if `reattach_cdroms` is set.
+	ReattachCDRom := s.Config.ReattachCDRom
+	if ReattachCDRom == 0 {
+		return multistep.ActionContinue
+	} else if ReattachCDRom < 1 || ReattachCDRom > 4 {
+		err := fmt.Errorf("'reattach_cdroms' should be between 1 and 4. if set to 0, `reattach_cdroms` is ignored and the step is skipped")
+		state.Put("error", fmt.Errorf("error reattach cdrom: %v", err))
+		return multistep.ActionHalt
+	}
```

Comment:
nit: Given that these are guard clauses we can drop the else. 
```suggestion
	if ReattachCDRom == 0 {
		return multistep.ActionContinue
	}
	if ReattachCDRom < 1 || ReattachCDRom > 4 {
		err := fmt.Errorf("'reattach_cdroms' should be between 1 and 4. if set to 0, `reattach_cdroms` is ignored and the step is skipped")
		state.Put("error", fmt.Errorf("error reattach cdrom: %v", err))
		return multistep.ActionHalt
	}
```

Reply:
Updated in 7563fce .